integreat_compass/cms/models/users/user.py

```diff
@@ -0,0 +1,135 @@
+    """
+
+    def create_user(
+        self, email, display_name, password=None, group=None, **extra_fields
+    ):
+        """
+        Create a new user and ensure they are added to the correct group (if any)
+
+        :param email: email address of the user, also used in lieu of username
+        :type email: str
+
+        :param display_name: name used for comments etc.
+        :type display_name: str
+
+        :param password: user password
+        :type password: str
+
+        :param group: either None or one of :attr:`~integreat_compass.cms.constants.roles.CHOICES`
+        :type group: str
+
+        :param extra_fields: additional fields
+        :type extra_fields: dict
+
+        :return: the newly created user
+        :rtype: ~integreat_compass.cms.models.users.user.User
+        """
+        email = self.normalize_email(email)
+        user = self.model(email=email, display_name=display_name, **extra_fields)
+
+        if group and group in roles.CHOICES:
+            user.groups.add(Group.objects.get(name=group))
+
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, display_name, password=None, **extra_fields):
+        """
+        Create a new super user
+
+        :param email: email address of the user, also used in lieu of username
+        :type email: str
+
+        :param display_name: name used for comments etc.
+        :type display_name: str
+
+        :param password: user password
+        :type password: str
+
+        :param extra_fields: additional fields
+        :type extra_fields: dict
+
+        :return: the newly created user
+        :rtype: ~integreat_compass.cms.models.users.user.User
+        """
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        return self.create_user(email, display_name, password, **extra_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin, AbstractBaseModel):
+    """
+    A custom User model that replaces the default Django User model.
+    """
+
+    email = models.EmailField(
+        unique=True,
+        verbose_name=_("email"),
+        help_text=_("Valid email address for this user"),
+    )
+    display_name = models.CharField(
+        max_length=128,
+        verbose_name=_("display name"),
+        help_text=_("This name will be shown as the author of votes, comments, etc."),
+    )
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+
+    objects = CustomUserManager()
+
+    USERNAME_FIELD = "email"
+    REQUIRED_FIELDS = ["display_name"]
+
+    @cached_property
+    def role(self):
+        """
+        We refer to Django user groups as roles.
+
+        :return: The role of this user
+        :rtype: ~integreat_compass.cms.models.users.role.Role
+        """
+        return self.groups.first().name if self.groups.exists() else None
+
```

Comment:
```suggestion
```
If we don't use the `Role` mode, I guess we can just reference `user.groups` directly?

Reply:
Hm... It might still be handy to have this method, maybe we can just rename it? Or is there a way around having to do `self.groups.first().name if self.groups.exists() else None` to get the role/group otherwise?

Reply:
Well, we could also just rely on the fact that `str(group)` is converted to `group.name` and just use `self.groups.first()` directly...

After all, `self.groups.first().name if self.groups.exists() else None` also makes two sql queries if the group exists, which is not really ideal... 

I can imagine that even this small property saves a little bit of typing:

```suggestion
    @cached_property
    def group(self):
        """
        Return the primary group of this user

        :return: The first group of this user
        :rtype: ~django.contrib.auth.models.Group
        """
        return self.groups.first()

```

Reply:
Well, we could also just rely on the fact that `str(group)` is converted to `group.name` and just use `self.groups.first()` directly...

After all, `self.groups.first().name if self.groups.exists() else None` also makes two sql queries if the group exists, which is not really ideal... 

I can imagine that even this small property saves a little bit of typing:

```suggestion
    @cached_property
    def group(self):
        """
        Return the primary group of this user

        :return: The first group of this user
        :rtype: ~django.contrib.auth.models.Group
        """
        return self.groups.first()

```

Reply:
Sounds good to me!